src/hotspot/cpu/x86/x86.ad

```diff
@@ -3972,7 +3973,6 @@ instruct fmaD_reg(regD a, regD b, regD c) %{
 
 // a * b + c
 instruct fmaF_reg(regF a, regF b, regF c) %{
-  predicate(UseFMA);
```

Comment:
You could add an assert to the encoding code. Just to ensure that we do not generate bad code, even if it is never executed during testing.

Reply:
Yes, thanks for your suggestion! Updated in the new commit.